### PR TITLE
Added EventService implementations to both Garden Pages for Admins

### LINKED ISSUE #517: Garden Admin page responsiveness
The garden admin pages should have event handlers that will update when things change on the server.

### PR DESCRIPTION
**Pull request by TheBurchLog**
Admin pages for Gardens now listen for events for live updates.

Ways to test:
1. Deploy child garden to see if it appears
2. Shut down child garden to see if status changes
3. Delete the Garden through the API and see if it is removed. 

Fixes #517

### CODE REVIEW

**Comment by TheBurchLog** [comment]
@hazmat345 

The destroy function has been added

File: src/ui/src/js/controllers/admin_garden.js [hunk 1]
```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
     loadGardens();
   };
 
-  EventService.addCallback('admin_system', (event) => {
+  EventService.addCallback('admin_garden', (event) => {
     switch (event.name) {
       case 'GARDEN_CREATED':
-        break;
+        $scope.data.push(event.payload);
+        $apply();
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden.js:60]: This is broken, generates an error.

File: src/ui/src/js/controllers/admin_garden.js [hunk 2]
```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
...
       case 'GARDEN_REMOVED':
-        break;
+        let new_data = []
+        for (var i = 0; i < $scope.data.length; i++) {
+            if ($scope.data[i].id != event.payload.id){
+                new_data.push($scope.data[i]);
+            }
+        }
+        $scope.data = new_data
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden.js:68]: This isn't necessary, you can pull an item out of an array using `splice`.

File: src/ui/src/js/controllers/admin_garden.js [hunk 3]
```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
...
 
-  EventService.addCallback('admin_system', (event) => {
+  EventService.addCallback('admin_garden', (event) => {
     switch (event.name) {
       case 'GARDEN_CREATED':
-        break;
+        $scope.data.push(event.payload);
+        $apply();
       case 'GARDEN_REMOVED':
-        break;
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden.js:61]: Why was this removed? `switch` statements needs `break`

File: src/ui/src/js/controllers/admin_garden.js [hunk 4]
```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
...
+            }
+        }
+        $scope.data = new_data
+      case 'GARDEN_UPDATED':
+        for (var i = 0; i < $scope.data.length; i++) {
+            if ($scope.data[i].id == event.payload.id){
+                $scope.data[i] = event.payload;
+            }
+        }
     }
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden.js:75]: Since you're modifying the contents of a scope data structure (rather than replacing it) add a `$scope.digest()` call at the end to update all the bindings.

File: src/ui/src/js/controllers/admin_garden_view.js [hunk 5]
```diff
@@ -79,6 +81,20 @@ export default function adminGardenViewController(
      }
   };
 
+  EventService.addCallback('admin_garden_view', (event) => {
+    switch (event.name) {
+      case 'GARDEN_UPDATED':
+        if ($scope.data.id == event.payload.id){
+            $scope.data = event.payload;
+
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden_view.js:89]: Whitespace

File: src/ui/src/js/controllers/admin_garden_view.js [hunk 6]
```diff
@@ -79,6 +81,20 @@ export default function adminGardenViewController(
...
+        }
```
Comment (hazmat345) [inline src/ui/src/js/controllers/admin_garden_view.js:90]: Should have a `break` here too

File: src/ui/src/js/controllers/admin_garden.js (hunk 3 above)
Comment (TheBurchLog) [reply src/ui/src/js/controllers/admin_garden.js:61]: let's just say it has been a while since someone has done switch statements...

File: src/ui/src/js/controllers/admin_garden.js (hunk 2 above)
Comment (TheBurchLog) [reply src/ui/src/js/controllers/admin_garden.js:68]: Yeah... that works as well. I kept reading up on `slice` instead of `splice` when I first did this. So that is why I avoided the array functions.

File: src/ui/src/js/controllers/admin_garden.js (hunk 4 above)
Comment (TheBurchLog) [reply src/ui/src/js/controllers/admin_garden.js:75]: $scope.disgest() did not resolve. Everything seems to now update dynamically after modifying the object in $scope.

File: src/ui/src/js/controllers/admin_garden.js (hunk 1 above)
Comment (TheBurchLog) [reply src/ui/src/js/controllers/admin_garden.js]: @hazmat345 Can you verify that you are still getting an error with this?

File: src/ui/src/js/controllers/admin_garden.js (hunk 4 above)
Comment (hazmat345) [reply src/ui/src/js/controllers/admin_garden.js:75]: Ehh, it should have been `$scope.$digest()`. But if things seem snappy without it then :shrug:
